kolibri/content/api.py

```diff
@@ -33,18 +33,18 @@ def title_description_filter(self, queryset, value):
 
     def filter_recommendations_for(self, queryset, value):
         recc_node = queryset.get(pk=value)
-        descendants = recc_node.get_descendants(include_self=False)
-        siblings = recc_node.get_siblings(include_self=False)
+        descendants = recc_node.get_descendants(include_self=False).exclude(kind__in=['topic', ''])
+        siblings = recc_node.get_siblings(include_self=False).exclude(kind__in=['topic', ''])
         data = descendants | siblings  # concatenates different querysets
         return data
 
     def filter_recommendations(self, queryset, value):
         # if ContentInteractionLog.objects.count() == 0:
-        content_ids = queryset.values_list('content_id', flat=True)
```

Comment:
Yes - any thoughts on a more efficient way of randomly selecting 25 content nodes?